mwdb/web/src/components/OAuth.js

```diff
@@ -0,0 +1,133 @@
+import React, { useContext, useEffect, useState } from "react";
+import { useHistory } from "react-router";
+
+import queryString from "query-string";
+
+import { APIContext } from "@mwdb-web/commons/api/context";
+import { AuthContext } from "@mwdb-web/commons/auth";
+import { View, getErrorMessage } from "@mwdb-web/commons/ui";
+
+export function OAuthLogin() {
+    const api = useContext(APIContext);
+    const [error, setError] = useState();
+    const [providers, setProviders] = useState([]);
+    const [chosenProvider, setChosenProvider] = useState();
+
+    async function getProviders() {
+        try {
+            const response = await api.axios.get("/oauth");
+            setProviders(response.data["providers"]);
+        } catch (e) {
+            setError(e);
+        }
+    }
+
+    async function login(provider) {
+        try {
+            const response = await api.axios.post(
+                `/oauth/${provider}/authenticate`
+            );
+            let expirationTime = new Date();
+            expirationTime.setTime(expirationTime.getTime() + 5 * 60 * 1000);
+            sessionStorage.setItem(
+                `openid_${response.data["state"]}`,
+                JSON.stringify({
+                    provider: provider,
+                    nonce: response.data["nonce"],
+                    action: "login",
+                    expiration: expirationTime,
+                })
+            );
```

Comment:
I think we should not pass Date objects to `JSON.stringify`. Let's use epoch representation instead.

Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

```suggestion
            const expirationTime = Date.now() + 5 * 60 * 1000;
            sessionStorage.setItem(
                `openid_${response.data["state"]}`,
                JSON.stringify({
                    provider: provider,
                    nonce: response.data["nonce"],
                    action: "login",
                    expiration: expirationTime,
                })
            );
```